core/unit_test/tools/TestEventCorrectness.hpp

```diff
@@ -395,12 +408,76 @@ TEST(kokkosp, parallel_scan) {
         }
         return MatchDiagnostic{true};
       });
-// Currently, this test is known to fail with OpenMPTarget
-#ifndef KOKKOS_ENABLE_OPENMPTARGET
   ASSERT_TRUE(success);
-#else
-  (void)success;
+}
+
+TEST(kokkosp, parallel_scan_no_fence) {
+  // FIXME_THREADS
+#ifdef KOKKOS_ENABLE_THREADS
+  if (std::is_same<Kokkos::DefaultExecutionSpace, Kokkos::Threads>::value)
+    GTEST_SKIP() << "skipping since the Thread backend always fences";
 #endif
+    // FIXME_OPENMPTARGET
+#ifdef KOKKOS_ENABLE_OPENMPTARGET
+  if (std::is_same<Kokkos::DefaultExecutionSpace,
+                   Kokkos::Experimental::OpenMPTarget>::value)
+    GTEST_SKIP()
+        << "skipping since the OpenMPTarget backend has unexpected fences";
+#endif
+
+  using namespace Kokkos::Test::Tools;
+  listen_tool_events(Config::DisableAll(), Config::EnableKernels(),
+                     Config::EnableFences());
+  auto success = validate_absence(
+      [=]() {
+        TestScanFunctor tf;
+        Kokkos::parallel_scan("dogs", Kokkos::RangePolicy<>(0, 1), tf);
+      },
+      [=](BeginFenceEvent begin_event) {
+        if (begin_event.name.find("Debug Only Check for Execution Error") !=
+                std::string::npos ||
+            begin_event.name.find("Kokkos Profile Tool Fence") !=
+                std::string::npos)
+          return MatchDiagnostic{false};
+        else
+          return MatchDiagnostic{true};
+      });
+  ASSERT_TRUE(success);
+}
+
+TEST(kokkosp, parallel_scan_no_fence_view) {
+  // FIXME_THREADS
+#ifdef KOKKOS_ENABLE_THREADS
+  if (std::is_same<Kokkos::DefaultExecutionSpace, Kokkos::Threads>::value)
+    GTEST_SKIP() << "skipping since the Thread backend always fences";
+#endif
+    // FIXME_OPENMPTARGET
+#ifdef KOKKOS_ENABLE_OPENMPTARGET
+  if (std::is_same<Kokkos::DefaultExecutionSpace,
+                   Kokkos::Experimental::OpenMPTarget>::value)
+    GTEST_SKIP()
+        << "skipping since the OpenMPTarget backend has unexpected fences";
+#endif
+
+  using namespace Kokkos::Test::Tools;
+  listen_tool_events(Config::DisableAll(), Config::EnableKernels(),
+                     Config::EnableFences());
+  Kokkos::View<typename TestScanFunctor::value_type> v("scan_result");
+  auto success = validate_absence(
+      [=]() {
+        TestScanFunctor tf;
+        Kokkos::parallel_scan("dogs", Kokkos::RangePolicy<>(0, 1), tf, v);
+      },
+      [=](BeginFenceEvent begin_event) {
+        if (begin_event.name.find("Debug Only Check for Execution Error") !=
```

Comment:
Are these fences still possible and acceptable here?

Reply:
Yes, we fence after every kernel if `KOKKOS_ENABLE_DEBUG_BOUNDS_CHECK` is enabled and we don't care about fences that come from the use of tools.